io-engine/src/grpc/v0/mayastor_grpc.rs

```diff
@@ -219,23 +220,23 @@ impl From<LvsError> for tonic::Status {
         match e {
             LvsError::Import {
                 source, ..
-            } => match source {
+            } => match source.to_errno() {
                 Errno::EINVAL => Status::invalid_argument(e.to_string()),
                 Errno::EEXIST => Status::already_exists(e.to_string()),
                 _ => Status::invalid_argument(e.to_string()),
             },
             LvsError::RepCreate {
                 source, ..
             } => {
-                if source == Errno::ENOSPC {
+                if source.to_errno() == Errno::ENOSPC {
                     Status::resource_exhausted(e.to_string())
                 } else {
                     Status::invalid_argument(e.to_string())
                 }
             }
             LvsError::RepDestroy {
                 source, ..
-            } => match source {
+            } => match source.to_errno() {
```

Comment:
Yeah, see example here, where ENOENT  != ENOMEDIUM
If not present already we should add a test on the BDD.

Reply:
See my comment above.